### PR TITLE
Add ec hints

### LINKED ISSUE #201: EC Hints
This requires implementing a basic `EcPoint` type and some methods to negate points, double them, and other operations. [Reference](https://github.com/lambdaclass/cairo-vm/blob/main/vm/src/hint_processor/builtin_hint_processor/secp/ec_utils.rs)

List:

- `EC_NEGATE`
- `EC_NEGATE_EMBEDDED_SECP`
- `EC_DOUBLE_SLOPE_V1`
- `COMPUTE_SLOPE_V1`
- `EC_DOUBLE_ASSIGN_NEW_X_V1`
- `EC_DOUBLE_ASSIGN_NEW_X_V2`
- `EC_DOUBLE_ASSIGN_NEW_X_V3`
- `EC_DOUBLE_ASSIGN_NEW_X_V4`
- `FAST_EC_ADD_ASSIGN_NEW_X`
- `FAST_EC_ADD_ASSIGN_NEW_X_V2`
- `FAST_EC_ADD_ASSIGN_NEW_X_V3`
- `FAST_EC_ADD_ASSIGN_NEW_Y`
- `EC_MUL_INNER`

### PR DESCRIPTION
**Pull request by mmsc2**

### CODE REVIEW

File: pkg/hints/ec_hint.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+	for i := 0; i < 3; i++ {
+		felt := val.Limbs[i]
+		signed := felt.ToSigned()
+		shifed := new(big.Int).Lsh(signed, uint(i*86))
+		sum.Add(sum, shifed)
+	}
+	return *sum
+}
+
+func FromBaseAddr(addr memory.Relocatable, virtual_machine vm.VirtualMachine) (BigInt3, error) {
```
Comment (fmoletta) [inline pkg/hints/ec_hint.go]:
```suggestion
func BigInt3FromBaseAddr(addr memory.Relocatable, virtual_machine vm.VirtualMachine) (BigInt3, error) {
```
Name is not descriptive enough and will clash with other types as we use an universal package for hints

File: pkg/hints/ec_hint.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+		if err == nil {
+			limbs = append(limbs, felt)
+		} else {
+			return BigInt3{}, errors.New("Identifier has no member")
+		}
+	}
+	return BigInt3{Limbs: limbs}, nil
+}
+
+func FromVarName(name string, virtual_machine vm.VirtualMachine, ids_data hint_utils.IdsManager) (EcPoint, error) {
```
Comment (fmoletta) [inline pkg/hints/ec_hint.go]:
```suggestion
func BigInt3FromVarName(name string, virtual_machine vm.VirtualMachine, ids_data hint_utils.IdsManager) (EcPoint, error) {
```
See previous comment

File: pkg/hints/ec_hint.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+	return EcPoint{X: x, Y: y}, nil
+}
+
+/*
+Implements main logic for `EC_NEGATE` and `EC_NEGATE_EMBEDDED_SECP` hints
+*/
+func ec_negate(virtual_machine vm.VirtualMachine, exec_scopes types.ExecutionScopes, ids_data hint_utils.IdsManager, secp_p big.Int) error {
+	point, err := ids_data.GetRelocatable("point", &virtual_machine)
+	fmt.Println("point ", point)
```
Comment (fmoletta) [inline pkg/hints/ec_hint.go]: Remove

**Review by fmoletta — COMMENTED** [review]
Most of the code in this PR uses snake_case, please fix it to use camelCase instead

**Review by fmoletta — APPROVED** [review]

**Review by jrchatruc — APPROVED** [review]

**Pull request merged**